### ISSUE TITLE
[Bug]  Failed to write plot header with error: 22 

### ISSUE THREAD

**Issue by eliliam** [Chia-Network/chia-blockchain#16956]
### What happened?

I have tried to create compressed plots on my GPU(Tesla P4) using the hybrid disk/ram mode that requires only 128GB of system memory on Ubuntu 22 on all of the versions of Bladebit(the one integrated in the `chia` command as well as the standalone version), however every single time after it allocates buffers it errors out with: 
```
 Failed to write plot header with error: 22.
 ```
 
 The commands I have ran are as follows:
 ```
# Chia command
chia plotters bladebit cudaplot -r 48 -c {CONTRACT_ADDR} -d /mnt/plots -v --compress 7 --disk-128 -t /mnt/chiaTemp -2
/mnt/chiaTemp

# Bladebit standalone 3.1.0 downloaded from releases page 
./bladebit_cuda -z 7 -f {FARMER_PUB_KEY} -c {CONTRACT_ADDR} --verbose cudaplot --disk-128 -t1 /mnt/chiaTemp/ /mnt/plots/
```

Am I doing anything wrong here? I find literally zero results when Googling for the error 22 in relation to Bladebit, any help would be greatly appreciated.

### Version

2.1.1

### What platform are you using?

Linux

### What ui mode are you using?

CLI

### Relevant log output

```shell
Bladebit Chia Plotter
Version      : 3.1.0
Git Commit   : e9836f8bd963321457bc86eb5d61344bfb76dcf0
Compiled With: gcc 9.4.0

[Global Plotting Config]
 Will create 1 plots.
 Thread count          : 48
 Warm start enabled    : false
 NUMA disabled         : false
 CPU affinity disabled : false
 Farmer public key     : {FARMER_PUB_KEY}
 Pool contract address : {CONTRACT_ADDR}
 Compression Level     : 7
 Benchmark mode        : disabled

[Bladebit CUDA Plotter]
 Host RAM            : 152 GiB
 Plot checks         : disabled

Selected cuda device 0 : Tesla P4
 CUDA Compute Capability   : 6.1
 SM count                  : 20
 Max blocks per SM         : 32
 Max threads per SM        : 2048
 Async Engine Count        : 2
 L2 cache size             : 2.00 MB
 L2 persist cache max size : 0.00 MB
 Stack Size                : 1.00 KB
 Memory:
  Total                    : 7.93 GB
  Free                     : 7.82 GB

Allocating buffers (this may take a few seconds)...
Kernel RAM required       : 89669060304  bytes ( 85515.08  MiB or 83.51  GiB )
Intermediate RAM required : 1642143744   bytes ( 1566.07   MiB or 1.53   GiB )
Host RAM required         : 28420603904  bytes ( 27104.00  MiB or 26.47  GiB )
Total Host RAM required   : 118089664208 bytes ( 112619.08 MiB or 109.98 GiB )
GPU RAM required          : 6163050496   bytes ( 5877.54   MiB or 5.74   GiB )
Allocating buffers...
Done.

Generating plot 1 / 1: c38f16733cd4bd2f9298eb2824227ba83b5c56789705bbfe61bfac7f21d86466
Plot temporary file: /mnt/plots/plot-k32-c07-2023-11-28-22-37-c38f16733cd4bd2f9298eb2824227ba83b5c56789705bbfe61bfac7f21d86466.plot.tmp

STDERR:

STDERR: Fatal Error:

STDERR: Failed to write plot header with error: 22.
```

**Comment by wjblanke** [comment]
can u create files in /mnt/plots? 22 is usually a permissions issue

**Comment by eliliam** [comment]
Yes, I have read/full write access on both /mnt/plots and /mnt/chiaTemp

**Comment by eliliam** [comment]
I just tested the GigaHorse plotter using the corresponding values in both and I am able to successfully plot with that, which leads me to think this is an actual issue with the bladebit plotter and not with the arguments I passed to the command.

**Comment by eliliam** [comment]
@wjblanke do you have any other suggestions or advice here? I need to start plotting and would rather use bladebit over gigahorse, but I can't seem to get bladebit to work.

**Comment by eliliam** [comment]
@wjblanke or @harold-b do either of you know what error 22 indicates? It's been difficult to find any documentation on these error codes.

**Comment by harold-b** [comment]
Error `22` is `EINVAL` (Invalid argument). What type of file system are you writing to?
You might need to disable direct-IO (should be able to do so from the CLI) depending on the target file system.

**Comment by harold-b** [comment]
I think no direct-IO for the output plot might not be exposed via `chia plotters` but if you use the `bladebit` executable directly then you should be able to specify `--no-direct-io` as a global argument to disable using direct I/O on the output directory to see if that's the issue.

**Comment by eliliam** [comment]
@harold-b thanks for the quick response! I am running this in an x86_64 Ubuntu 22.04 VM, the disks in question are SSDs directly passed through from Proxmox(the host) to the VM and then merged via `mdadm` to create a RAID0 array of them. Could you explain a bit more about direct-io? I'm trying to identify if it's an issue with the disk passthrough or if it is due to the `mdadm` array.

I will go try using the binary itself instead of the `chia plotters` cli, but seeing as this is an issue that impacts some users, it would be really nice to have all the same functionality exposed via both the `bladebit` executable and the `chia plotters` CLI.

**Comment by harold-b** [comment]
`13` is "permission denied". See if the directory has the correct permissions set up that the process running `bladebit_cuda` is able to write to it

**Comment by eliliam** [comment]
I have good news, after fixing permissions and using the `--no-direct-io` flag, things seem to be working great! My only suggestion at this point is to add in all the missing options that the `bladebit` executable has into the `chia plotters` version, since it makes things a lot easier without requiring you to specify your farmer pub key or contract address.

**Comment by eliliam** [comment]
@wjblanke do you have any other suggestions or advice here? I need to start plotting and would rather use bladebit over gigahorse, but I can't seem to get bladebit to work.

**Comment by harold-b** [comment]
Thanks for the feedback. Perhaps we'll get something even better put together
